### DISCUSSION TITLE
How to parse tokens into different text

### DISCUSSION THREAD

**Discussion by lthoerner** [pest-parser/pest#932] [Q&A]
I am writing a parser for my shell which needs to do two main things:
1. Parse a line of input to a tree structure
2. Parse certain tokens into different strings (a good example of this is that a `"\\n"` needs to be turned into a `"\n"`)

Is there any way that I could do the second thing here without having to iterate over every parsed char after the whole thing has been parsed?

**Comment by Tartasprint** [comment]
May be you can create a rule for your token, and create a rule `my_any` that you can use in place of `ANY` in the other rules:
```pest
token_new_line = {"\\n"}
my_any = { token_new_line | ANY }
```
You may need to adapt this, for example maybe it's not ANY but some other rule that you need. But I guess there is a pattern here.
